### PR TITLE
Reduce unnecessary `Arg` creation in happy paths

### PR DESCRIPTION
**Pull request by carterkozak**
==COMMIT_MSG==
Reduce unnecessary `Arg` creation in happy paths
==COMMIT_MSG==

### CODE REVIEW

File: dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/JacksonEmptyContainerLoader.java [hunk 1]
```diff
@@ -138,16 +137,19 @@ private static int decrement(int maxRecursion, TypeMarker<?> originalType) {
...
+            for (Method method : clazz.getMethods()) {
+                if (Modifier.isStatic(method.getModifiers())
+                        && method.getParameterCount() == 1
+                        && method.getAnnotation(JsonCreator.class) != null) {
+                    return method;
+                }
+            }
         }
-        return Optional.empty();
+        return null;
```
Comment (carterkozak) [inline dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/JacksonEmptyContainerLoader.java:152]: removed unnecessary optional and and stream.

File: dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/JacksonEmptyContainerLoader.java [hunk 2]
```diff
@@ -95,10 +93,11 @@
     }
 
     private static int decrement(int maxRecursion, TypeMarker<?> originalType) {
-        Preconditions.checkState(
-                maxRecursion > 0,
-                "Unable to construct an empty instance as @JsonCreator requires too much recursion",
-                SafeArg.of("type", originalType));
```
Comment (carterkozak) [inline dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/JacksonEmptyContainerLoader.java:101]: This still appears to show up heavily in JFRs

**Review by schlosna — APPROVED** [review]

File: dialogue-core/src/main/java/com/palantir/dialogue/core/ConcurrencyLimitedChannel.java [hunk 3]
```diff
@@ -131,8 +131,10 @@ public String toString() {
                 int uriIndex,
                 CautiousIncreaseAggressiveDecreaseConcurrencyLimiter limiter,
                 TaggedMetricRegistry taggedMetrics) {
-            Preconditions.checkArgument(
-                    uriIndex != -1, "uriIndex must be specified", SafeArg.of("channel-name", channelName));
+            if (uriIndex == 1) {
```
Comment (schlosna) [inline dialogue-core/src/main/java/com/palantir/dialogue/core/ConcurrencyLimitedChannel.java]: typo

```suggestion
            if (uriIndex == -1) {
```

Comment (carterkozak) [reply dialogue-core/src/main/java/com/palantir/dialogue/core/ConcurrencyLimitedChannel.java]: :+1:

**Pull request merged**